### PR TITLE
Refactored Unit Test Startup and Teardown

### PR DESCRIPTION
**Pull request by maxcapodi78**
Now EDB tests and all tests that doesn't need AEDT runs before the others and outside the desktop

### CODE REVIEW

File: _unittest/conftest.py [hunk 1]
```diff
@@ -85,27 +85,37 @@ def my_setup(self):
...
+                try:
+                    aedtapp.close_project(None, False)
+                except:
+                    pass
+            del self.aedtapps
 
     def add_app(self, project_name=None, design_name=None, solution_type=None, application=None):
+        if "oDesktop" not in dir(sys.modules["__main__"]):
+            if self.aedtapps:
+                self.aedtapps[0].release_desktop()
```
Comment (Alberto-DM) [inline _unittest/conftest.py:115]: Is this a safety mechanism? 
I don't get when it will execute this release_desktop

**Review by Alberto-DM — COMMENTED** [review]
Good improvement for test stability.
Made some comments on unclear parts.

File: _unittest/conftest.py [hunk 2]
```diff
@@ -172,14 +182,14 @@ def setup(self):
 
 @pytest.fixture(scope="session", autouse=True)
 def desktop_init():
-    desktop = Desktop(desktop_version, non_graphical, new_thread)
-    desktop.disable_autosave()
-    yield desktop
-
+    # desktop = Desktop(desktop_version, non_graphical, new_thread)
+    # desktop.disable_autosave()
+    # yield desktop
```
Comment (MaxJPRey) [inline _unittest/conftest.py:187]:
```suggestion
```

File: _unittest/conftest.py [hunk 3]
```diff
@@ -172,14 +182,14 @@ def setup(self):
...
+    yield
     # If new_thread is set to false by a local_config, then don't close the desktop.
     # Intended for local debugging purposes only
```
Comment (MaxJPRey) [inline _unittest/conftest.py:190]:
```suggestion
```

File: _unittest/conftest.py [hunk 4]
```diff
@@ -172,14 +182,14 @@ def setup(self):
...
-    if new_thread or os.name == "posix":
-        desktop.close_desktop()
+    # if new_thread or os.name == "posix":
+    #    desktop.close_desktop()
```
Comment (MaxJPRey) [inline _unittest/conftest.py:192]: @maxcapodi78 It is not required anymore to have this new-thread for local debugging session?
```suggestion
```

**Review by MaxJPRey — APPROVED** [review]

File: _unittest/conftest.py (hunk 1 above)
Comment (maxcapodi78) [reply _unittest/conftest.py]: you are right. it should be in the teardown method.

File: _unittest/conftest.py [hunk 5]
```diff
@@ -85,27 +85,37 @@ def my_setup(self):
...
-            try:
-                edbapp.close_edb()
-            except:
-                pass
-        del self.edbapps
-        for aedtapp in self.aedtapps[::-1]:
+    def my_teardown(self, close_desktop=False):
+        if close_desktop and not is_ironpython:
+            d = Desktop(desktop_version, non_graphical, False)
+            d.release_desktop()
```
[the remark this answers is not in the document]
Comment (maxcapodi78) [reply _unittest/conftest.py]: done. Thanks

File: _unittest/conftest.py (hunk 4 above)
Comment (maxcapodi78) [reply _unittest/conftest.py]: Thanks

**Pull request merged**